manifests/install.pp

```diff
@@ -40,9 +40,9 @@
 
   if !$::vault::disable_mlock {
     exec { "setcap cap_ipc_lock=+ep ${vault_bin}":
```

Comment:
I like the idea behind this PR, it should be a maintained state, not a one off...  There is a nice alternative to doing this in an exec though...

https://forge.puppet.com/stm/file_capability

@cruwe / @jsok what do you think about using that?  It looks like a nicer way to me.